src/System.Private.CoreLib/shared/System/Reflection/Emit/StringToken.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Reflection.Emit
+{
+    public struct StringToken
+    {
+        internal StringToken(int str)
+        {
+            Token = str;
+        }
+
+        // Returns the metadata token for this particular string.  
+        // Generated by a call to Module.GetStringConstant().
+        //
+        public int Token { get; }
+
+        public override int GetHashCode() => Token;
+
+        public override bool Equals(object obj) => obj is StringToken st && Equals(st);
```

Comment:
The IL and resulting asm for `obj is SomeValueType svt && Equals(svt)` has been worse than `obj is SomeValueType && Equal((SomeValueType)obj)`.  I'm told that it's improved in the C# 8 compiler, but I've not verified.  I don't know whether these StringTokens show up on fast paths, but if they do, it'd be good to validate that this has no negative impact on code quality.

Reply:
@stephentoub I don't think object like Equal overload is on the hot path, secondly, all other *Token.cs implementations have the same pattern.